packages/trader/src/Stores/Modules/Trading/Helpers/process.ts

```diff
@@ -0,0 +1,55 @@
+import { ContractType as ContractTypeHelper } from 'Stores/Modules/Trading/Helpers/contract-type';
+import * as ContractType from '../Actions/contract-type';
+import * as Duration from '../Actions/duration';
+import * as StartDate from '../Actions/start-date';
+import { TTradeStore } from 'Types';
+
+const processInSequence = async (
+    store: TTradeStore,
+    functions: ReturnType<typeof getMethodsList> | ReturnType<typeof getExpiryMethodsList>
+) => {
+    //@ts-expect-error need to typescript convert base store and make parameters optional else it will break
```

Comment:
@henry-deriv let's use assertions (as ) or a type guard at the place of error instead of this `//@ts-expect-error` comment.

Reply:
i already tried @maryia-deriv

the issue is `getSnapshot` expects an input of `string[]`. So i tried getSnapshot(['']) to get rid of the error but breaks the application. So i cant do anything until getSnapshot is ts migrated to make its parameters optional.